Document life of a connection and event flow (#233)
autokitteh/autokitteh: docs/life_of_an_event.md

> # Life of an Event

Reviewer: This page would benefit a lot from a flow diagram below the steps. You can create one with markdown - both in GH and Docusaurus I think.

> 1. For example, an external provider sends some data to an integration.

Reviewer: ```suggestion
1. An external provider sends data to an integration asynchronously.
```
